tests/PHPStan/Type/TypeToPhpDocNodeTest.php

```diff
@@ -290,12 +302,22 @@ public function dataToPhpDocNodeWithoutCheckingEquals(): iterable
 
 		yield [
 			new ConstantFloatType(2.35),
-			'2.35000000000000008882',
+			'2.35',
 		];
 
 		yield [
 			new ConstantFloatType(100),
-			'100',
+			'100.0',
+		];
+
+		yield [
+			new ConstantFloatType(8.202343767574732),
+			'8.202343767574732',
+		];
+
+		yield [
+			new ConstantFloatType(1e80),
+			'1.0E+80',
```

Comment:
phpdoc parser needs to be fixed to parse numbers in scientific format, at least the ones with `.` where the type is unambiguous